src/azure-cli/azure/cli/command_modules/role/_params.py

```diff
@@ -87,9 +87,7 @@ def load_arguments(self, _):
         c.argument('scopes', nargs='+')
         c.argument('role', completer=get_role_definition_name_completion_list)
         c.argument('skip_assignment', arg_type=get_three_state_flag(),
-                   help='Skip creating the default assignment, which allows the service principal to access resources under the current subscription. '
-                        'When specified, --scopes will be ignored. You may use `az role assignment create` to create '
-                        'role assignments for this service principal later.')
+                   deprecate_info=c.deprecate(target='--skip-assignment', hide=True), help='No-op.')
```

Comment:
`--skip-assignment` is now a no-op argument to keep the command backward compatible.